mesonbuild/build.py

```diff
@@ -1737,6 +1732,20 @@ def process_vs_module_defs_kw(self, kwargs: T.Dict[str, T.Any]) -> None:
                 'a file object, a Custom Target, or a Custom Target Index')
         self.process_link_depends(path)
 
+    def extract_targets_as_list(self, kwargs: T.Dict[str, T.Any], key: str) -> T.List[LibTypes]:
+        bl_type = self.environment.coredata.get_option(OptionKey('default_both_libraries'))
+        if bl_type == 'auto':
+            bl_type = 'static' if isinstance(self, StaticLibrary) else 'shared'
+
+        def _resolve_both_libs(lib: LibTypes) -> LibTypes:
+            if isinstance(lib, BothLibraries):
+                return getattr(lib, bl_type)
```

Comment:
The use of `getattr` here is going to confuse the type-checker. I think it would be good to just add a method to `BothLibraries` that take the `Literal['static', 'shared', 'auto']` and retuns the value.